src/chainparams.cpp

```diff
@@ -27,6 +27,20 @@ void ReadSigNetArgs(const ArgsManager& args, CChainParams::SigNetOptions& option
         }
         options.challenge.emplace(ParseHex(signet_challenge[0]));
     }
+    if (args.IsArgSet("-signetblocktime")) {
+        const auto signetblocktime = args.GetArgs("-signetblocktime");
+        if (signetblocktime.size() != 1) {
+            throw std::runtime_error(strprintf("%s: -signetblocktime cannot be multiple values.", __func__));
+        }
+        if (!args.IsArgSet("-signetchallenge")) {
+            throw std::runtime_error(strprintf("%s: -signetblocktime cannot be set without -signetchallenge", __func__));
+        }
```

Comment:
Looks like it is only networking settings in there and all of them only print error logs, I think we'd want to explicitly throw in this case

Reply:
You're right, we do want to throw. `AppInitParameterInteraction()` does seem to raise `InitError` based on bad parameter interaction. I'm not sure why we have both `AppInitParameterInteraction()` and `InitParameterInteraction()` and which needs to be used when, it just seems like since this effectively is parameter interaction we'd like to raise that somewhere in here. Hopefully someone else can chime in?